src/scanner-cli.ts

```diff
@@ -51,20 +52,22 @@ export function normalizePlatformName(): 'windows' | 'linux' | 'macosx' {
 /**
  * Where to download the SonarScanner CLI
  */
-function getScannerCliUrl(properties: ScannerProperties, version: string): URL {
+function getScannerCliUrl(properties: ScannerProperties, version: SemVer): URL {
   // Get location to download scanner-cli from
 
   // Not in default to avoid populating it for non scanner-cli users
   const scannerCliMirror = properties[ScannerProperty.SonarScannerCliMirror] ?? SCANNER_CLI_MIRROR;
+  const arch = version.compare('6.1.0') >= 0 ? '-x64' : '';
   const scannerCliFileName =
-    'sonar-scanner-cli-' + version + '-' + normalizePlatformName() + '.zip';
+    'sonar-scanner-cli-' + version + '-' + normalizePlatformName() + arch + '.zip';
   return new URL(scannerCliFileName, scannerCliMirror);
 }
 
 export async function downloadScannerCli(properties: ScannerProperties): Promise<string> {
-  const version = properties[ScannerProperty.SonarScannerCliVersion] ?? SCANNER_CLI_VERSION;
-  if (!/^[\d.]+$/.test(version)) {
-    throw new Error(`Version "${version}" does not have a correct format."`);
+  const versionProp = properties[ScannerProperty.SonarScannerCliVersion] ?? SCANNER_CLI_VERSION;
+  const version = semver.coerce(versionProp);
```

Comment:
`semver.coerce('1.2.3.4567')` looses the build number information, so this needs to be in `getScannerCliUrl` to ensure we build the URL including the build number

Reply:
yep, I fixed that after seeing the tests failures